quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java

```diff
@@ -207,6 +209,26 @@ FeatureBuildItem getFeature() {
         return new FeatureBuildItem("keycloak");
     }
 
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    ConfigBuildItem initConfig(KeycloakRecorder recorder) {
+        Config.init(new MicroProfileConfigProvider());
+        recorder.initConfig();
+        return new ConfigBuildItem();
+    }
+
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    @Consume(ConfigBuildItem.class)
+    ProfileBuildItem configureProfile(KeycloakRecorder recorder) {
+        Profile profile = Profile.configure(
+                new QuarkusProfileConfigResolver(),
+                new PropertiesFileProfileConfigResolver()); // Need profile.properties for now as testsuite relies on it
```

Comment:
We can do a follow-up but it should be possible to avoid installing the `PropertiesFileProfileConfigResolver` to the distribution but only during tests.

This can be done using a conditional build step like here https://github.com/keycloak/keycloak/blob/1b7ae48dcb915d1cff39d530d2b1466f32604d20/quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java#L426.

Reply:
Wonder if it wouldn't be nice for folks to have the option of using `conf/profile.properties`. Plan is to extend more on the profile stuff in the future to leverage it more, and could be nice to be able to separate out that part from the other config perhaps?

Reply:
My only argument against it is that we have now a configuration schema and we should rely on it whenever possible to configure the server.

For now, I would not introduce another file but just rely on the existing configuration sources (cli, keycloak.conf, etc)